service-base/src/test/java/org/eclipse/hono/service/credentials/BaseCredentialsServiceTest.java

```diff
@@ -11,142 +11,176 @@
+        final String invalidTimestamp = "yakshaver";
 
-        assertFalse(service.containsStringValueForField(testData, TEST_FIELD));
+        final JsonObject testData = createValidCredentialsObject();
+        final JsonObject firstSecret = testData.getJsonArray(FIELD_SECRETS).getJsonObject(0);
+        firstSecret.put(FIELD_SECRETS_NOT_BEFORE, invalidTimestamp);
+        
+        final Message<JsonObject> msg = createMessageMockForPayload(testData);
+        service.processCredentialsMessage(msg);
+        
+        verify(msg).reply(resultWithStatusCode(HTTP_BAD_REQUEST));
     }
 
     @Test
-    public void testContainsStringValueForFieldWhenPresent() {
-        JsonObject testData = new JsonObject();
-        testData.put(TEST_FIELD, "yakshaver");
+    public void testCredentialsAddWithMissingSecrets() {
+        final JsonObject testData = createValidCredentialsObject();
 
-        assertTrue(service.containsStringValueForField(testData, TEST_FIELD));
+        testData.remove(FIELD_SECRETS);
+        
+        final Message<JsonObject> msg = createMessageMockForPayload(testData);
+        service.processCredentialsMessage(msg);
+        
+        verify(msg).reply(resultWithStatusCode(HTTP_BAD_REQUEST));
     }
 
     @Test
-    public void testContainsStringValueForFieldWhenPresentEmpty() {
-        JsonObject testData = new JsonObject();
-        testData.put(TEST_FIELD, "");
+    public void testCredentialsAddWithNoSecrets() {
+        final JsonObject testData = createValidCredentialsObject();
+        testData.put(FIELD_SECRETS, new JsonArray());
 
-        assertFalse(service.containsStringValueForField(testData, TEST_FIELD));
+        final Message<JsonObject> msg = createMessageMockForPayload(testData);
+        service.processCredentialsMessage(msg);
+        
+        verify(msg).reply(resultWithStatusCode(HTTP_BAD_REQUEST));
     }
 
     @Test
-    public void testContainsValidSecretValueWithMissingSecret() {
-        JsonObject testData = new JsonObject();
+    public void testCredentialsAddWithEmptySecret() {
+        final JsonObject testData = createValidCredentialsObject();
+        
+        JsonArray secrets = new JsonArray();
+        secrets.add(new JsonObject());
+        
+        testData.put(FIELD_SECRETS, secrets);
 
-        assertFalse(service.containsValidSecretValue(testData));
+        final Message<JsonObject> msg = createMessageMockForPayload(testData);
+        service.processCredentialsMessage(msg);
+        
+        verify(msg).reply(resultWithStatusCode(HTTP_CREATED));
     }
 
-    @Test
-    public void testContainsValidSecretValueWithNoSecrets() {
-        JsonObject testData = new JsonObject();
-        JsonArray testSecrets = new JsonArray();
-        testData.put(FIELD_SECRETS, testSecrets);
+    private JsonObject resultWithStatusCode(int statusCode) {
+        final JsonObject result = new JsonObject();
+        result.put(MessageHelper.APP_PROPERTY_TENANT_ID, TEST_TENANT);
```

Comment:
Please use RequestResponseApiConstants.FIELD_TENANT_ID instead (recent change to eliminate the "_" in json objects).
This (previous) key crashes the test in my environment.